src/app/components/shared/items/selectable-items/selectable-items.component.spec.ts

```diff
@@ -1,23 +1,120 @@
-import { ComponentFixture, TestBed } from "@angular/core/testing";
-import { SelectableItemsComponent } from "./selectable-items.component";
+import { createComponentFactory, Spectator } from "@ngneat/spectator";
+import { IconsModule } from "@shared/icons/icons.module";
+import {
+  ISelectableItem,
+  SelectableItemsComponent,
+} from "./selectable-items.component";
 
 describe("SelectableItemsComponent", () => {
-  let component: SelectableItemsComponent<any>;
-  let fixture: ComponentFixture<SelectableItemsComponent<any>>;
+  let spec: Spectator<SelectableItemsComponent<string>>;
+  let selectionChangeSpy: jasmine.Spy;
 
-  beforeEach(() => {
-    TestBed.configureTestingModule({
-      imports: [SelectableItemsComponent],
-    }).compileComponents();
+  const testOptions: ISelectableItem<string>[] = [
+    { label: "Option 1", value: "value1" },
+    { label: "Option 2", value: "value2" },
+    { label: "Option 3", value: "value3" },
+  ];
+
+  const createComponent = createComponentFactory({
+    component: SelectableItemsComponent<string>,
+    imports: [IconsModule],
+  });
 
-    fixture = TestBed.createComponent(SelectableItemsComponent);
-    component = fixture.componentInstance;
-    fixture.detectChanges();
+  function getButtons(): HTMLButtonElement[] {
+    return spec.queryAll<HTMLButtonElement>("button");
+  }
+
+  beforeEach(() => {
+    spec = createComponent({ detectChanges: false });
+    selectionChangeSpy = spyOn(spec.component.selectionChange, "emit");
   });
 
   it("should create", () => {
-    expect(component).toBeTruthy();
+    spec.detectChanges();
+    expect(spec.component).toBeTruthy();
+  });
+
+  describe("selectionChange event", () => {
+    it("should emit selectionChange when a non-selected item is clicked", () => {
+      spec.setInput("options", testOptions);
+      spec.setInput("selection", "value1");
+      spec.detectChanges();
+
+      const buttons = getButtons();
+      // Click on the second button (value2)
+      buttons[1].click();
+      spec.detectChanges();
+
+      expect(selectionChangeSpy).toHaveBeenCalledOnceWith("value2");
+    });
+
+    it("should not emit selectionChange when an already selected item is clicked", () => {
+      spec.setInput("options", testOptions);
+      spec.setInput("selection", "value1");
+      spec.detectChanges();
+
+      const buttons = getButtons();
+      // Click on the first button (value1) which is already selected
+      buttons[0].click();
+      spec.detectChanges();
```

Comment:
Use the `clickButton` test helper


```suggestion
      clickButton(spec, getButtons()[0])
```